File: javascript/leetcode/448. Find All Numbers Disappeared in an Array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDisappearedNumbers = function (nums) {
  const result = [];

  nums.forEach((el, pos) => {
    let ind = Math.abs(el) - 1;
    if (nums[ind] > 0) nums[ind] *= -1;
  });

  nums.forEach((el, pos) => {
    if (el > 0) result.push(pos + 1);
  });

  return result;
};

File: c++/leetcode/448. Find All Numbers Disappeared in an Array.cpp
class Solution
{
public:
  vector<int> findDisappearedNumbers(vector<int> &nums)
  {
    vector<int> result;

    for (int i = 0; i < nums.size(); i++)
    {
      const int ind = abs(nums[i]) - 1;
      if (nums[ind] > 0)
        nums[ind] *= -1;
    }

    for (int i = 1; i <= nums.size(); i++)
    {
      if (nums[i - 1] > 0)
        result.push_back(i);
    }

    return result;
  }
};